include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -0,0 +1,778 @@
+ *
+ * \returns The value of the specified configuration.
+ *
+ * \details
+ *
+ * This function has the same semantics as std::get.
+ *
+ * ### Complexity
+ *
+ * Constant.
+ *
+ * ### Exception
+ *
+ * No-throw guarantee.
+ *
+ * ### Concurrency
+ *
+ * Thread-safe.
+ */
+template <size_t elem_no, detail::config_element_concept ... configs_t>
+constexpr auto const & get(detail::configuration<configs_t...> const & cfg) noexcept
+{
+    using type_list_type = detail::tuple_type_list_t<typename detail::configuration<configs_t...>::base_type>;
+
+    static_assert(elem_no < meta::size<type_list_type>::value,
+                  "Index out of range.");
+    return std::get<elem_no>(cfg).value;
+}
+
+/*!\brief Returns the value of the corresponding config at the specified position.
+ * \tparam     elem_no   Non-type template parameter specifying the position of the config to get.
+ * \tparam     configs_t Template parameter pack specifying the seqan3::detail::config_elements.
+ * \param[in]  cfg       The configuration to query for it's config.
+ *
+ * \returns The value of the specified configuration.
+ *
+ * \details
+ *
+ * This function has the same semantics as std::get.
+ *
+ * ### Complexity
+ *
+ * Constant.
+ *
+ * ### Exception
+ *
+ * No-throw guarantee.
+ *
+ * ### Concurrency
+ *
+ * Thread-safe.
+ */
+template <size_t elem_no, detail::config_element_concept ... configs_t>
+constexpr auto && get(detail::configuration<configs_t...> && cfg) noexcept
+{
+    using type_list_type = detail::tuple_type_list_t<typename detail::configuration<configs_t...>::base_type>;
+
+    static_assert(elem_no < meta::size<type_list_type>::value,
+                  "Index out of range.");
+    return std::get<elem_no>(std::move(cfg)).value;
+}
+
+/*!\brief Returns the value of the corresponding config at the specified position.
+ * \tparam     elem_no   Non-type template parameter specifying the position of the config to get.
+ * \tparam     configs_t Template parameter pack specifying the seqan3::detail::config_elements.
+ * \param[in]  cfg       The configuration to query for it's config.
+ *
+ * \returns The value of the specified configuration.
+ *
+ * \details
+ *
+ * This function has the same semantics as std::get.
+ *
+ * ### Complexity
+ *
+ * Constant.
+ *
+ * ### Exception
+ *
+ * No-throw guarantee.
+ *
+ * ### Concurrency
+ *
+ * Thread-safe.
+ */
+template <size_t elem_no, detail::config_element_concept ... configs_t>
+constexpr auto const && get(detail::configuration<configs_t...> const && cfg) noexcept
+{
+    using type_list_type = detail::tuple_type_list_t<typename detail::configuration<configs_t...>::base_type>;
+
+    static_assert(elem_no < meta::size<type_list_type>::value,
+                  "Index out of range.");
+
+    // TODO: It is unclear why return must be wrapped in std::move here.
+    return std::move(std::get<elem_no>(std::move(cfg)).value);
+}
+} // namespace seqan3
```

Comment:
So if I understand this correctly, `std::get<0>` gives my the config_element and `seqan3::get<0>` gives me the cargo?

Reply:
I don’t like this but yes. Alternatively, the get could be a member function of the configuration and the global get is never overloaded. Because the deferred configs shouldn’t be accessed  directly. For the algorithm you want to check if a specific config is set but you are only interested in the cargo. Got alignment  we have a specific enum based get interface that I can only offer as global function.